### PR TITLE
[stable/rabbitmq-ha] added network policy

### PR DESCRIPTION
**Pull request by aksenk**
#### What this PR does / why we need it:
This PR adds network policy template to the chart.

#### Special notes for your reviewer:
Network policy is disabled by default.
If we just enable netpol, than we allow cross-pod communication for cluster nodes in statefulset.
If we enable allowClientLabel, we allow external connects from pods with label "rabbitmq-client: true".
If we need more flexibility, we can add any custom config to our configuration.

This is my vision of a better path for network policies.
If you have any comments or suggestions, I am ready to listen to them.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by zanhsieh** [comment]
@steven-sheehy
Could you LGTM this PR please? Thank you.

File: stable/rabbitmq-ha/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "rabbitmq-ha.fullname" . }}
+  labels:
+    app: {{ include "rabbitmq-ha.fullname" . }}
+    name: {{ include "rabbitmq-ha.name" . }}
+    fullname: {{ include "rabbitmq-ha.fullname" . }}
+    chart: {{ include "rabbitmq-ha.chart" . }}
+    app.kubernetes.io/name: {{ include "rabbitmq-ha.fullname" . }}
```
Comment (steven-sheehy) [inline stable/rabbitmq-ha/templates/networkpolicy.yaml:11]: Don't mix and match old labels and new. We should match what the rest of the chart uses:
```yaml
  labels:
    app: {{ template "rabbitmq-ha.name" . }}
    chart: {{ template "rabbitmq-ha.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
```

File: stable/rabbitmq-ha/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - protocol: TCP
+      port: 15675
+    {{- end }}
+    {{- if .Values.rabbitmqAmqpsSupport.enabled }}
+    - protocol: TCP
+      port: 5671
+    {{- end }}
+    {{- if .Values.rabbitmqPrometheusPlugin.enabled }}
+    - protocol: TCP
+      port: {{ .Values.rabbitmqPrometheusPlugin.port }}
```
Comment (steven-sheehy) [inline stable/rabbitmq-ha/templates/networkpolicy.yaml:62]: I would prefer the named port being used for these instead of the integer port

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**